In [4]:
import pandas as pd, numpy as np

In [7]:
df = pd.read_csv('estimate_olist_vs_our_estimate.csv')

In [8]:
df

,shipping_duration_days,my_estimation,olist_estimate
0,7,14,15
1,16,16,18
2,7,15,21
3,6,10,11
4,25,15,40
...,...,...,...
108616,17,20,24
108617,9,14,17
108618,4,13,17
108619,1,8,10


In [9]:
import pandas as pd
from sklearn.metrics import mean_absolute_error, mean_squared_error
import numpy as np

# Assuming df is your DataFrame
y_true = df['shipping_duration_days']
my_pred = df['my_estimation']
olist_pred = df['olist_estimate']

# Calculate MAE
mae_my = mean_absolute_error(y_true, my_pred)
mae_olist = mean_absolute_error(y_true, olist_pred)

# Calculate RMSE
rmse_my = mean_squared_error(y_true, my_pred, squared=False)
rmse_olist = mean_squared_error(y_true, olist_pred, squared=False)

# Print results
print(f"📦 My Estimation - MAE: {mae_my:.3f}, RMSE: {rmse_my:.3f}")
print(f"🏷️ Olist Estimate - MAE: {mae_olist:.3f}, RMSE: {rmse_olist:.3f}")

📦 My Estimation - MAE: 6.204, RMSE: 8.426
🏷️ Olist Estimate - MAE: 12.913, RMSE: 15.304


# The Results - use our model ! 

### `My Estimation` - MAE: 6.204 - so the prediction are off by { +  `6` } | { - `6` } days  